### DISCUSSION TITLE
How to pass "rust-analyzer.trace.server": "verbose"

### DISCUSSION THREAD

**Discussion by xzbdmw** [mrcjkb/rustaceanvim#377] [Q&A]
This seems to be an vscode setting, I can't make it work.
Do you know any logging that can see response time during request?

**Comment by mrcjkb** [comment]
Hey 👋 

You should be able to put that in the `server["rust-analyzer"` table.
See [advanced configuration](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#gear-advanced-configuration) in the readme.

**Reply by tvsfx** [reply]
FYI, I did set it up like this, but I saw no additional logging happening, and I think I've found the reason:
https://github.com/search?q=repo%3Arust-lang%2Frust-analyzer%20trace.server&type=code
This setting seems to be caught client-side by the VSCode extension, and will hence not do anything server-side.
I was hoping this setting would use `window/logMessage` messages or something similar, but I guess it doesn't.

**Reply by mrcjkb** [reply]
If it's a vscode plugin setting, then it won't do anything in rustaceanvim.
If it's related to client side logging, you probably want to use `:h vim.lsp.log` in Neovim.

**Reply by tvsfx** [reply]
yes, I just mentioned it for posterity :) I've been using the following in my LSP setup to somewhat emulate the VSCode functionality:
```lua
            vim.lsp.set_log_level("trace")
            vim.lsp.log.set_format_func(vim.inspect)
```
(although this applies to all LSPs)
